### PR TITLE
Fix: Allow Helion to adjust MC amount when buying cards during Research

### PR DESCRIPTION
**Pull request by nwai90**
Ref: https://github.com/bafolts/terraforming-mars/issues/1000

Screenshot 2020-07-16 at 5 36 34 PM

(Checking by title to avoid introducing too many changes)

### CODE REVIEW

File: src/components/PaymentWidgetMixin.ts [hunk 1]
```diff
@@ -36,12 +37,19 @@ export const PaymentWidgetMixin = {
...
             let currentValue: number = (this as any)[target];
             let maxValue: number = (this as any).player[target];
-            if (target === "megaCredits") maxValue = this.getMegaCreditsMax();
+            const isResearchPhase = (this as any).$data.isResearchPhase;
+
+            if (isResearchPhase && target === "megaCredits") {
+                maxValue = (this as any).player.cardCost * 4;
+            } else if (target === "megaCredits") {
+                maxValue = this.getMegaCreditsMax();
+            }
```
Comment (jijiqw) [inline src/components/PaymentWidgetMixin.ts:51]: globalevent may decrease players's mc up to 20 ,  cardCost * 4 may greater than player.megaCredits

Comment (nwai90) [reply src/components/PaymentWidgetMixin.ts:51]: GlobalEvents don't trigger during the Research phase. This change is specific to `player.runResearchPhase()`. `player.cardCost` is used to account for Terralabs and Polyphemos.

If player attempts to overspend MC an error will be thrown: `"You don't have that many mega credits"`

**Comment by nwai90** [comment]
@vincentneko @alrusdi @bafolts I found another bug on `master`, also fixed in this PR. If you choose to buy 0 cards as Helion during Research phase but decide to "spend" n heat, that heat gets incorrectly converted into MC.

**Comment by alrusdi** [comment]
@nwai90 is it ready to be merged? I don't see any problems

**Comment by nwai90** [comment]
It should be ready; I removed the `wip` label.

Issues like overpayment and underpayment are already handled in existing code. I think we can improve it as we go along.

File: src/components/PaymentWidgetMixin.ts (hunk 1 above)
Comment (jijiqw) [reply src/components/PaymentWidgetMixin.ts:51]: It's Ok ,thanks

**Pull request merged**